packages/swing-store/src/importer.js

```diff
@@ -47,158 +40,109 @@ export async function importSwingStore(exporter, dirPath = null, options = {}) {
-        const vatInfo = vatArtifacts.get(vatID) || {};
-        if (tag === 'snapshot') {
-          // `export.snapshot.{vatID}.current` directly identifies the current snapshot artifact
-          vatInfo.snapshotKey = value;
-        } else if (tag === 'transcript') {
-          // `export.transcript.${vatID}.current` contains a metadata record for the current
-          // state of the current transcript span as of the time of export
-          const metadata = JSON.parse(value);
-          vatInfo.transcriptKey = artifactKey(tag, vatID, metadata.startPos);
-          artifactMetadata.set(vatInfo.transcriptKey, metadata);
-        }
-        vatArtifacts.set(vatID, vatInfo);
-      } else {
-        artifactMetadata.set(artifactKey(tag, vatID, pos), JSON.parse(value));
-      }
+    } else if (value == null) {
+      allMetadata.delete(key);
     } else {
-      Fail`unknown artifact type tag ${q(tag)} on import`;
+      allMetadata.set(key, value);
     }
   }
 
-  // At this point we should have acquired the entire KV store state, plus
-  // sufficient metadata to identify the complete set of artifacts we'll need to
-  // fetch along with the information required to validate each of them after
-  // fetching.
-  //
-  // Depending on how the export was parameterized, the metadata may also include
-  // information about historical artifacts that we might or might not actually
-  // fetch depending on how this import was parameterized
-
-  // Fetch the set of current artifacts.
-
-  // Keep track of fetched artifacts in this set so we don't fetch them a second
-  // time if we are trying for historical artifacts also.
-  const fetchedArtifacts = new Set();
-
-  for await (const [vatID, vatInfo] of vatArtifacts.entries()) {
-    // For each vat, we *must* have a transcript span.  If this is not the very
-    // first transcript span in the history of that vat, then we also must have
-    // a snapshot for the state of the vat immediately prior to when the
-    // transcript span begins.
-    vatInfo.transcriptKey ||
-      Fail`missing current transcript key for vat ${q(vatID)}`;
-    const transcriptInfo = artifactMetadata.get(vatInfo.transcriptKey);
-    transcriptInfo || Fail`missing transcript metadata for vat ${q(vatID)}`;
-    let snapshotInfo;
-    if (vatInfo.snapshotKey) {
-      snapshotInfo = artifactMetadata.get(vatInfo.snapshotKey);
-      snapshotInfo || Fail`missing snapshot metadata for vat ${q(vatID)}`;
-    }
-    if (!snapshotInfo) {
-      transcriptInfo.startPos === 0 ||
-        Fail`missing current snapshot for vat ${q(vatID)}`;
+  // Now take each metadata record and install the stub/pruned entry
+  // into the DB.
+
+  for (const [key, value] of allMetadata.entries()) {
+    const [tag, ...pieces] = key.split('.');
+    if (tag === 'bundle') {
+      const bundleID = pieces[0];
+      internal.bundleStore.addBundleRecord(bundleID);
+    } else if (tag === 'snapshot') {
+      const [_vatID, endPos] = pieces;
```

Comment:
Yeah, I suppose.. I do want this whole section to remain short enough to fit on the screen at once, and doing a lot of extra validation distracts from the primary logic ("read tag, dispatch to a store"). 

The export-data *keys* exist to:
* 1: give us a handle, so we can modify or delete the same record later (e.g. when we append to a transcript span, we need to tell the exportCallback to change the previous record's value, increasing `endPos` and changing `hash`)
* 2: allow a sorted dump of the export-data to tell a useful story to a human

The duplication of fields between key and value happens because it would be super-annoying to omit that data from the value. The importing code would have to synthesize/merge a record out of some parsed bits of the key, and the parsed bits of the value. And a human reader of the export-data would have to look at both columns to figure out what was going on. I agree that it would remove the error/mismatch cases ("normal form" usually does), but I suspect it would be a net loss for legibility.

The import process doesn't really need much out of the keys. It needs the tag so it can dispatch the *value* to the right store, but it could do its job with just the tag and the value. I wanted to reflect that truth in this loop.

The bundleID is one exception: the value is pure bundle, and we need the key to hold ID. I could tidy that up by changing `addBundleRecord` to take `(key, value)`, I suppose. Passing `JSON.parse(value)` into the `add*Record` methods makes their function signatures nicer to read, but if I wanted to stick to this "just tag and value" story I guess I could change their signatures to `(key, value)` too.

The `if (!!key.endsWith('.current' ..` check seemed like a good idea, but it also detracts from this story.

So, some options are:
* 1: remove the validation from this loop, pass `(key, value)` to each `add*Record` function
  * this might argue for changing the method names, since we aren't giving them a record, and giving them key+value means they're tied more closely to the import process (specifically tied to the format of export-data), so maybe `importBundleRecord`
* 2: leave it as-is: no more validation, no less
* 3: add more validation: `pieces.length`, parse the value, assert that vatID matches in key and value, assert that startPos or endPos or snapPos matches

I'm inclined to stick with 2, but let me know what you feel about it.

Reply:
I'm actually thinking a plain pass-through would be good.

Reply:
Ok, I'll rewrite this with option 1: do just enough parsing to figure out which store owns the record, then pass through to e.g. `snapStore.importSnapshotRecord(key, value)`.

Reply:
pushed, in 821c41dd73622cd090170bfbaad663b55167d8e7